### PR TITLE
domain: use dedicated lock for expiredTimeStamp4PC

### PR DESCRIPTION
**Pull request by aidendou**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/45400

### What is changed and how it works?

Added a dedicated lock for domain.expiredTimeStamp4PC so it doesn't share the lock with domain, therefore schema reload (domain.Reload()) won't be blocked by comipler.Compile().

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: domain/domain_test.go [hunk 1]
```diff
@@ -19,6 +19,9 @@ import (
 	"crypto/tls"
 	"encoding/json"
 	"fmt"
+	"github.com/pingcap/tidb/parser/mysql"
```
Comment (hawkingrei) [inline domain/domain_test.go:22]: Please use ```goimports``` to format it.

Comment (hawkingrei) [reply domain/domain_test.go:22]: run ```make bazel_prepare``` to update bazel config.

Comment (aidendou) [reply domain/domain_test.go]: thx @hawkingrei. not sure if it's my goland setting, goimports didn't format the file the way CI wanted. manually fixed.

File: domain/BUILD.bazel [hunk 2]
```diff
@@ -134,13 +134,16 @@ go_test(
...
+        "//parser/mysql",
         "//parser/terror",
         "//server",
         "//session",
+        "//sessionctx/stmtctx",
         "//sessionctx/variable",
         "//store/mockstore",
         "//testkit",
         "//testkit/testsetup",
+        "//types",
```
Comment (AndreMouche) [inline domain/BUILD.bazel:146]: why?

**Review by AndreMouche — COMMENTED** [review]
Rest LGTM

**Review by AndreMouche — APPROVED** [review]
LGTM

**Comment by AndreMouche** [comment]
@zimulala @qw4990  PTAL, thanks!

File: domain/BUILD.bazel (hunk 2 above)
Comment (hawkingrei) [reply domain/BUILD.bazel:146]: It is the bazel config. it need to update the dependency to speed up build/linter/UT.

**Review by jackysp — APPROVED** [review]
LGTM

**Pull request merged**